src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,75 @@
-import React from 'react';
+import { useEffect, useCallback } from 'react';
+import { Todo } from '../../types/Todo';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from '../../app/store';
+import { useAppSelector } from '../../hooks/useAppSelector';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
+
+interface TodoFilterProps {
+  todos: Todo[];
+  handleFilterChange: (filteredTodos: Todo[]) => void;
+}
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  todos,
+  handleFilterChange,
+}) => {
+  const dispatch = useDispatch<AppDispatch>();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleFilterTodos = () => {
+    let filteredTodos = todos.filter(todo => {
+      switch (status) {
+        case 'active':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+
+    if (query) {
+      filteredTodos = filteredTodos.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    handleFilterChange(filteredTodos);
+  };
+
+  useEffect(() => {
+    handleFilterTodos();
+  }, [query, todos, status]);
+
+  const handleChangeFilter = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      dispatch(filterSlice.actions.setStatus(event.target.value as Status));
```

Comment:
Ensure that the `setStatus` action in `filterSlice` is correctly defined and that the `Status` type includes all possible status values ('all', 'active', 'completed').